components/kyma-environment-broker/internal/process/provisioning/apply_kyma_step.go

```diff
@@ -0,0 +1,108 @@
+package provisioning
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process/steps"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	kymaResourceNamespace = "kcp-system"
+)
+
+type ApplyKymaStep struct {
+	operationManager *process.OperationManager
+}
+
+var _ process.Step = &ApplyKymaStep{}
+var kymaGVK = schema.GroupVersionKind{Group: "operator.kyma-project.io", Version: "v1alpha1", Kind: "Kyma"}
+
+func NewApplyKymaStep(os storage.Operations) *ApplyKymaStep {
+	return &ApplyKymaStep{operationManager: process.NewOperationManager(os)}
+}
+
+func (a *ApplyKymaStep) Name() string {
+	return "Apply_Kyma"
+}
+
+func (a *ApplyKymaStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if operation.K8sClient == nil {
+		return a.operationManager.OperationFailed(operation, "operation does not contain initialized k8s client", nil, logger)
+	}
+	k8s := operation.K8sClient
```

Comment:
isn't `operation.K8sClient` a client to SKR's kube-api? I thought the kyma resource would live in the same cluster as kyma control plane. At the moment https://github.com/kyma-project/control-plane/pull/2216 is implemented to save the kubeconfig to KCP so the lifecycle manager can get the kubeconfig to SKR.